src/rgw/rgw_auth.cc

```diff
@@ -53,6 +53,16 @@ rgw_auth_transform_old_authinfo(req_state * const s)
     int get_perm_mask() const {
       return perm_mask;
     }
+
+    std::string to_str() const {
```

Comment:
I think it would be better to avoid stringstream in anything like a common path, at least if we don't rely on our own stream operator overrides.  thoughts?